annubes/tasks/task.py

```diff
@@ -1,209 +1,211 @@
+        noise_std: Standard deviation of input noise.
+            Defaults to 0.01.
+
+    Raises:
+        ValueError: if `catch_prob` is not between 0 and 1.
+    """
+
+    name: str
+    session: dict[str, float] = field(default_factory=lambda: {"v": 0.5, "a": 0.5})
+    stim_intensities: list[float] = field(default_factory=lambda: [0.8, 0.9, 1])
+    catch_prob: float = 0.5
+    catch_intensity: float = 0
+    shuffle_trials: bool = True
+    max_sequential: int | None = None
+    n_outputs: int = 2
+    output_intensities: list[float] = field(default_factory=lambda: [0, 1])
+    stim_time: int = 1000
+    fix_intensity: float = 0
+    fix_time: int | None = 100
+    input_baseline: float = 0.2
+    delay: int = 0
+    dt: int = 20
+    tau: int = 100
+    noise_std: float = 0.01
+    rescaling_coeff: float = 0
+
+    def __post_init__(self):
+        if not self.catch_prob >= 0 and self.catch_prob < 1:
+            msg = "`catch_prob` must be between 0 and 1."
+            raise ValueError(msg)
+
+        sum_session_vals = sum(self.session.values())
+        for i in self.session:
+            self.session[i] = self.session[i] / sum_session_vals
+        if not self.shuffle_trials:
+            self.session = OrderedDict(self.session)
 
         # Derived attributes
-        self.trials = {}
-        self.trials["name"] = self.name
-        self.modalities = list(OrderedDict.fromkeys(char for string in session_in for char in string))
-        self.modality_idx = {m: i for i, m in enumerate(self.modalities)}
-        self.n_in = len(self.modalities) + 1  # +1 for start cue
-        self.value_in.sort()
-        self.imin = self.value_in[0]
-        self.imax = self.value_in[-1]
-        self.coeff = 0.6
-        if (self.value_fixation is not None) and self.scaling:
-            self.value_fixation = self._scale_input(self.value_fixation, self.coeff, self.imin, self.imax)
-        self.T = self.inter_trial + self.t_fixation + self.t_in
-        self.t = np.linspace(0, self.T, self.dt)
-        self.t = np.linspace(0, self.T, int((self.T + self.dt) / self.dt))
-        self.value_out.sort()
-        self.low_out = self.value_out[0]
-        self.high_out = self.value_out[1]
-
-        # Checks
-        # session_in
-        tolerance = 1 / 50
-        if not abs(sum(self.session_in.values()) - 1) < tolerance:
-            raise ValueError("The sum of the probabilities of `session_in` must be 1.")
-        # catch_prob
-        if not (catch_prob >= 0 and catch_prob < 1):
-            raise ValueError("`catch_prob` must be higher or equal to 0, or lower than 1.")
-
-    def _scale_input(
+        self.modalities = list(dict.fromkeys(char for string in self.session for char in string))
+        self.n_inputs = len(self.modalities) + 1  # includes start cue
+        trial_duration = self.delay + self.fix_time + self.stim_time
+        self.t = np.linspace(0, trial_duration, int((trial_duration + self.dt) / self.dt))  # TODO: rename attribute
```

Comment:
Is this TODO still needed? No strong opinion, self.t is also already fine for me.

Reply:
I made it `self.time`